plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java

```diff
@@ -249,8 +250,8 @@ private void registerModel(MLRegisterModelInput registerModelInput, ActionListen
             mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
                 String taskId = response.getId();
                 mlTask.setTaskId(taskId);
-                mlModelManager.registerMLModel(registerModelInput, mlTask);
-                listener.onResponse(new MLRegisterModelResponse(taskId, MLTaskState.CREATED.name()));
+                System.out.println("mlModelManager calls registerMLRemoteModel");
+                mlModelManager.registerMLRemoteModel(registerModelInput, mlTask, listener);
```

Comment:
Is this the expectation that, only remote model registration will be asynchronous? If yes, can we please add a comment here also?

Reply:
Just the opposite, remote model registration is synchronous only so the model_id can be returned right away. Comments will be added in the next PR.